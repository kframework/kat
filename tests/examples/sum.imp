
int n , s ;
n = 10 ;
s = 0 ;
while (0 <= n) {
  n = n + -1 ;
  s = s + n ;
}
