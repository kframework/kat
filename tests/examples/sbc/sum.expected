Result ==K <kat-imp>
  <s>
    #STUCK ( ) ~> #compile-result .Rules , < <imp>
      <k>
        int n , s , .Ids ; n = 10 ; while ( 0 <= n ) { n = n + -1 ; s = s + n ; }
      </k>
      <mem>
        .Map
      </mem>
    </imp> --> <imp>
      <k>
        if ( 0 <= n ) { { n = n + -1 ; s = s + n ; } while ( 0 <= n ) { n = n + -1 ; s = s + n ; } } else { }
      </k>
      <mem>
        n |-> 10
        s |-> V0
      </mem>
    </imp> > , < <imp>
      <k>
        false ~> if ( 0 <= n ) { { n = n + -1 ; s = s + n ; } while ( 0 <= n ) { n = n + -1 ; s = s + n ; } } else { }
      </k>
      <mem>
        n |-> V1
        s |-> V2
      </mem>
    </imp> --> <imp>
      <k>
        .
      </k>
      <mem>
        n |-> V1
        s |-> V2
      </mem>
    </imp> > , < <imp>
      <k>
        true ~> if ( 0 <= n ) { { n = n + -1 ; s = s + n ; } while ( 0 <= n ) { n = n + -1 ; s = s + n ; } } else { }
      </k>
      <mem>
        n |-> V1
        s |-> V2
      </mem>
    </imp> --> <imp>
      <k>
        if ( 0 <= n ) { { n = n + -1 ; s = s + n ; } while ( 0 <= n ) { n = n + -1 ; s = s + n ; } } else { }
      </k>
      <mem>
        n |-> V1 +Int -1
        s |-> V2 +Int ( V1 +Int -1 )
      </mem>
    </imp> >
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        if ( 0 <= n ) { { n = n + -1 ; s = s + n ; } while ( 0 <= n ) { n = n + -1 ; s = s + n ; } } else { }
      </k>
      <mem>
        n |-> V3
        s |-> V4
      </mem>
    </imp>
  </harness>
</kat-imp>
